packages/project-access/src/file/findFiles.ts

```diff
@@ -1,20 +1,59 @@
-import { basename, dirname, sep } from 'path';
+import type { Editor } from 'mem-fs-editor';
+import { basename, dirname, join, sep } from 'path';
 import { default as find } from 'findit2';
-import findUp from 'find-up';
+import { existsSync as exists } from 'fs';
+
+/**
+ * Add missing dump properties
+ */
+declare module 'mem-fs-editor' {
+    type FileMap = { [key: string]: { state: 'modified' | 'deleted' } };
+
+    export interface Editor {
+        dump(cwd: string): FileMap;
+    }
+}
+
+/**
+ * Search for 'filename' starting from 'root' in the list of virtual (not committed) files. Returns array of paths that contain the file.
+ *
+ * @param files - - array of paths that contain the filename found on the filesystem
+ * @param filename - filename to search
+ * @param root - root folder to start search
+ * @param stopFolders - list of folder names to exclude (search doesn't traverse into these folders)
+ * @param fs - mem-fs-editor instance
+ * @returns - enhanced array of paths that contain the filename
+ */
+function checkVirtualFiles(files: string[], filename: string, root: string, stopFolders: string[], fs: Editor) {
```

Comment:
Could we just return the result of search on virtual file system and do the merge of 
`[...physicalFSresults, ...virtualFSresults]` outside in the consuming function `findFiles()`. I think this would make the scope of this function cleaner.

Reply:
I am not in favor of separating it, since we have that patter of the option `fs` parameter all over the code, I would also like to consistently keep it here.

Reply:
Guess there is a misunderstanding, Intention was not to remove the `fs` argument. What I meant is changing the function to:

```typescript
function findVirtualFiles(filename: string, root: string, stopFolders: string[], fs: Editor): string[]
```

So we would return the result array and not pass it into the function and fill it, same signature as `findFiles()`.

Reply:
The function does a little more than just finding virtual files, it also removes files that were found on the filesystem but are removed in the `mem-fs`. How about renaming it to `applyChangesFromMemory`?

Reply:
Ah, got it. In this case it is ok. I thought the find functionality on virtual files might also be a candidate for standalone usage at some point. We can also keep it like this and extract find in case there is demand later on.